File: leetcode/heap/0215_kth_largest_element.cpp
// LC 215. Kth Largest Element in an Array | Medium
#include <vector>
#include <queue>
using namespace std;

class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> pq;
        for (int n : nums) { pq.push(n); if (pq.size() > k) pq.pop(); }
        return pq.top();
    }
};

File: leetcode/heap/0215_kth_largest_element.py
# LC 215. Kth Largest Element in an Array | Medium
import heapq

class Solution:
    def findKthLargest_sort(self, nums, k): return sorted(nums, reverse=True)[k-1]
    def findKthLargest(self, nums, k): return heapq.nlargest(k, nums)[-1]
